test_scripts/Smoke/API/037_RegisterAppInterface_PositiveCase.lua

```diff
@@ -0,0 +1,119 @@
+---------------------------------------------------------------------------------------------------
+-- User story: Smoke
+-- Use case: RegisterAppInterface
+-- Item: Happy path
+--
+-- Requirement summary:
+-- [RegisterAppInterface] SUCCESS: getting SUCCESS:RegisterAppInterface() during reregistration
+--
+-- Description:
+-- Mobile application sends valid RegisterAppInterface request after unregistration and
+-- gets RegisterAppInterface "SUCCESS" response from SDL
+
+-- Pre-conditions:
+-- a. HMI and SDL are started
+-- b. appID is registered and activated on SDL
+-- c. appID is currently in Background, Full or Limited HMI level
+
+-- Steps:
+-- appID requests RegisterAppInterface
+
+-- Expected:
+-- SDL checks if RegisterAppInterface is allowed by Policies
+-- SDL sends the BasicCommunication notification to HMI
+-- SDL responds with (resultCode: SUCCESS, success:true) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSmoke = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Local Variables ]]
+local requestParams = {
+	syncMsgVersion = {
+		majorVersion = 2,
+		minorVersion = 2,
+	},
+	appName = "SyncProxyTester",
+	ttsName = {
+		{
+			text ="SyncProxyTester",
+			type ="TEXT",
+		},
+	},
+	ngnMediaScreenAppName = "SPT",
+	vrSynonyms = {
+		"VRSyncProxyTester",
+	},
+	isMediaApplication = true,
+	languageDesired = "EN-US",
+	hmiDisplayLanguageDesired = "EN-US",
+	appHMIType = {
+		"DEFAULT",
+	},
+	appID = "123456",
+	deviceInfo = {
+		hardware = "hardware",
+		firmwareRev = "firmwareRev",
+		os = "os",
+		osVersion = "osVersion",
+		carrier = "carrier",
+		maxNumberRFCOMMPorts = 5
+	}
+}
+
+local function SetNotificationParams()
+	local notificationParams = {
+		application = {}
+	}
+	notificationParams.application.appName = requestParams.appName
+	notificationParams.application.ngnMediaScreenAppName = requestParams.ngnMediaScreenAppName
+	notificationParams.application.isMediaApplication = requestParams.isMediaApplication
+	notificationParams.application.hmiDisplayLanguageDesired = requestParams.hmiDisplayLanguageDesired
+	notificationParams.application.appType = requestParams.appHMIType
+	notificationParams.application.deviceInfo = {
+		name = "127.0.0.1",
+		id = config.deviceMAC,
+		transportType = "WIFI",
+		isSDLAllowed = true
+	}
+	notificationParams.application.policyAppID = requestParams.appID
+	notificationParams.ttsName = requestParams.ttsName
+	notificationParams.vrSynonyms = requestParams.vrSynonyms
+	return notificationParams
+end
+
+local allParams = {
```

Comment:
@GetmanetsIrina Bearing in mind the last comment from my side we can omit defining of ```allPArams``` variable here and to create 2 ones: ```requestParams``` and ```notificationParams```
This simplify code a little

Reply:
@dboltovskyi , Done in 607c20fb6767f2c4c0e34e662869ef978977e810.